tests/test_datatypes.py

```diff
@@ -30,10 +30,16 @@ class TypeMixin(db.TemporaryTable):
     null = "NOT NULL"
     record_count = 3
     extra_sql = None
+    extension_types = ["vector"]
 
     def test_type(self, conn, cursor, schema_table, data):
         bincopy = CopyManager(conn, schema_table, self.cols)
-        bincopy.copy(data)
+        try:
+            bincopy.copy(data)
+        except psycopg2.errors.UndefinedFile:
+            if any(t in self.extension_types for t in self.datatypes):
+                pytest.skip("Unsupported datatype")
+            raise
```

Comment:
Can you help me understand in what case this try/except block was needed?  It works for me without this.  What versions of postgresql, psycopg2, pytest are you using?

Reply:
That try/except block is triggered if I switch out pgvector/pgvector:pg12 for postgres:12 in docker-compose.yml.

Reply:
I can’t reproduce it.  When I try that, the error is caught [here](https://github.com/altaurog/pgcopy/blob/master/tests/conftest.py#L88-L89)

Reply:
This error is happening for me on my arm64 Mac. It does not happen on my amd64 linux machine. I suspect there is a difference between the upstream Docker containers for arm64 and amd64.

Reply:
whoa, thanks.

Reply:
I tried on arm64 linux, I get the same behavior as on amd64 linux.

Reply:
Rented a MacInCloud M2 Mac mini to ensure nothing in my local setup was interfering with results. Got the same error. I also printed out the container architecture to confirm the containers are linux/arm64.

![image](https://github.com/altaurog/pgcopy/assets/42753606/41405667-f2b3-4aab-a378-039d3b25e647)

So it appears to be a Mac compatibility issue rather than arm64 in general. Why I do not know. I have spent as much energy as I can on this.

Reply:
Thank you very much @wburklund.  This is certainly much more than can be expected but definitely very helpful.  GitHub actions was my first thought, but unfortunately [it is not possible to run docker on the arm64 mac runners](https://github.com/marketplace/actions/setup-docker-on-macos#arm64-processors-m1-m2-m3-series-used-on-macos-14-images-are-unsupported).  My theory is that the postgres error when trying to create the table has a different code for some reason and therefore doesn’t meet the [expected condition for unsupported datatype](https://github.com/altaurog/pgcopy/blob/vector/tests/conftest.py#L108-L109).  I would start by logging the `pgcode` in the except block to see if this is the case.  I can try MacInCloud myself or Amazon EC2 to experiment further, although I also have very limited time for this line of investigation.
I don’t mind just releasing as-is, with or without the additional error handling you introduced, but in any case I won’t be able to do anything on this project until later coming week at the earliest.

Reply:
I'd like to release as-is, with the additional error handling.

Reply:
I set up [tests on macos 14 arm64 github runners](https://github.com/altaurog/pgcopy/actions/runs/8433334740) with postgresql versions 12-16 and python 3.11; they are all passing.  If the issue is docker-related, I won’t be able to reproduce it on github, since docker isn’t available.

What version of libpq and psycopg2 are you using?

Reply:
Docker container is picking up psycopg2 version 2.9.9, which is bundled with PostgreSQL 16 libpq, according to the documentation.